### DISCUSSION TITLE
cpp2b - an experimental build system for cpp2

### DISCUSSION THREAD

**Discussion by zaucy** [hsutter/cppfront#1222] [Show and tell]
I wrote a build system for cpp2 called `cpp2b`. You can see the repository here: https://github.com/zaucy/cpp2b

It's written in cpp2 and builds itself!

The project itself is just for fun and to make it easier for me to try out cpp2. I don't expect it to be used seriously in any capacity.

[cpp2b-demo.webm](https://github.com/user-attachments/assets/4f0053db-fc13-4906-a2cd-e5e632c61f38)

It's inspired by the zig build system, but is nowhere even comparable in capability. The only similarity `cpp2b` has is there is a `build.cpp2` file which is used to configure your project.

**Comment by hsutter** [comment]
Cool, thanks for sharing!

**Comment by MaxSagebaum** [comment]
Looks nice.

**Comment by sookach** [comment]
Very cool, nice work.

**Comment by DyXel** [comment]
Looks very nice, you know, for a "for fun project" it actually looks quite elaborated :p

**Comment by vspefs** [comment]
Far as I know, one great thing about Zig build system is Zig standard library, which enables users to do a lot of cool things conveniently. Like you said, I don't think that kind of convenience can be easily copied.

If we're to push the "Cpp2 build system" thing harder, it's possible (and should be easy) to extend Zig build system so that it supports cpp2. Otherwise, it could be A LOT of work involved.

Inspiring work nevertheless. Using Cpp2 to write Cpp2 build system puts a good test to Cpp2 and our understanding of it.

**Reply by zaucy** [reply]
Yes! Specifically zigs `std.Build` might be a lot of work to replicate and I, by no means, plan to do that 😅.

For now I'm simply enjoying using cpp2 + cpp20 modules. I think my focus on the project will be mainly that and keep the `build.cpp2` part quite minimal.
